contracts/chain-adapters/Arbitrum_AdapterBase.sol

```diff
@@ -0,0 +1,180 @@
+     * @notice This contract must hold at least getL1CallValue() amount of ETH to send a message via the Inbox
+     * successfully, or the message will get stuck.
+     * @param target Contract on Arbitrum that will receive message.
+     * @param message Data to send to target.
+     */
+    function _relayMessage(address target, bytes memory message) internal {
+        uint256 requiredL1CallValue = _contractHasSufficientEthBalance(RELAY_MESSAGE_L2_GAS_LIMIT);
+
+        L1_INBOX.createRetryableTicket{ value: requiredL1CallValue }(
+            target, // destAddr destination L2 contract address
+            L2_CALL_VALUE, // l2CallValue call value for retryable L2 message
+            L2_MAX_SUBMISSION_COST, // maxSubmissionCost Max gas deducted from user's L2 balance to cover base fee
+            L2_REFUND_L2_ADDRESS, // excessFeeRefundAddress maxgas * gasprice - execution cost gets credited here on L2
+            L2_REFUND_L2_ADDRESS, // callValueRefundAddress l2Callvalue gets credited here on L2 if retryable txn times out or gets cancelled
+            RELAY_MESSAGE_L2_GAS_LIMIT, // maxGas Max gas deducted from user's L2 balance to cover L2 execution
+            L2_GAS_PRICE, // gasPriceBid price bid for L2 execution
+            message // data ABI encoded data of L2 message
+        );
+    }
+
+    /**
+     * @notice Bridge tokens to Arbitrum.
+     * @notice This contract must hold at least getL1CallValue() amount of ETH to send a message via the Inbox
+     * successfully, or the message will get stuck.
+     * @param l1Token L1 token to deposit.
+     * @param amount Amount of L1 tokens to deposit and L2 tokens to receive.
+     * @param to Bridge recipient.
+     */
+    function _relayTokens(
+        address l1Token,
+        address,
+        uint256 amount,
+        address to
+    ) internal {
+        // Check if this token is USDC, which requires a custom bridge via CCTP.
+        if (_isCCTPEnabled() && l1Token == address(usdcToken)) {
+            _transferUsdc(to, amount);
+        }
+        // If not, we can use the Arbitrum gateway
+        else {
+            uint256 requiredL1CallValue = _contractHasSufficientEthBalance(RELAY_TOKENS_L2_GAS_LIMIT);
+
+            // Approve the gateway, not the router, to spend the hub pool's balance. The gateway, which is different
+            // per L1 token, will temporarily escrow the tokens to be bridged and pull them from this contract.
+            address erc20Gateway = L1_ERC20_GATEWAY_ROUTER.getGateway(l1Token);
+            IERC20(l1Token).safeIncreaseAllowance(erc20Gateway, amount);
+
+            // `outboundTransfer` expects that the caller includes a bytes message as the last param that includes the
+            // maxSubmissionCost to use when creating an L2 retryable ticket: https://github.com/OffchainLabs/arbitrum/blob/e98d14873dd77513b569771f47b5e05b72402c5e/packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/gateway/L1GatewayRouter.sol#L232
```

Comment:
This seems like a dead link now. I _think_ the correct reference for that contract is: https://github.com/OffchainLabs/arbitrum-classic/blob/551a39b381dcea81e03e7599fcb01fddff4fe96c/packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/gateway/L1GatewayRouter.sol#L240

We are however referencing their legacy contracts though. They seem to be maintaining compatibility moving forward but I wonder if we should try to reference the current implementations instead.

Reply:
Nice catch https://github.com/across-protocol/contracts/pull/610/commits/05dfc6eefae26f129ab68223a011766894ec144b